CIP-AnyDataIsDatum/README.md

```diff
@@ -0,0 +1,45 @@
+---
+CIP: ????
+Title: Hash-Checked Data
+Author: Maksymilian 'zygomeb' Brodowicz <zygomeb@gmail.com>
+Status: Draft
+Type: Standards
+Created: 2022-10-25
+License: CC-BY-4.0
+---
+
+## Simple Summary / Abstract
+
+Hashing data on-chain is quite expensive from a computational perspective. Currently, however, we are forced to use the `serializeData` primitive combined with a hashing function of our choice. It would be computationally beneficial to let the node hash the data before running the smart contracts, giving us only a guarantee that this mapping is correct, just as we do with Datums presently.
+
+Therefore we propose extending the ScriptContext from just a mapping between DatumHash and Datum to a more universal mapping between BuiltinByteString and BuiltinData.
```

Comment:
the `ScriptContext` is not a mapping. I think you mean the `txInfoData` field of `TxInfo`.

Reply:
Yes, this sentence refers to the one field in ScriptContext. Although at the time of writing I was unsure whether or not to include the other way of doing this via including a new field `txInfoAdditionalData` to contain these which, to the other thread, would make it backwards compatible.